cumulus/zombienet/zombienet-sdk/tests/zombie_ci/elastic_scaling/pov_recovery.rs

```diff
@@ -0,0 +1,225 @@
+		.wait_log_line_count_with_timeout(
+			"set_validation_data inherent needs to be present in every block",
+			false,
+			LogLineCountOptions::no_occurences_within_timeout(Duration::from_secs(10)),
+		)
+		.await?;
+
+	assert!(result.success(), "Consensus hook failed at {}: {:?}", collator_elastic.name(), result);
+
+	// Wait (up to 10 seconds) until pattern occurs more than 35 times
+	let options = LogLineCountOptions {
+		predicate: Arc::new(|n| n > 35),
+		timeout: Duration::from_secs(10),
+		wait_until_timeout_elapses: false,
+	};
+
+	let name = "recovery-target";
+	log::info!("Ensuring blocks are imported using PoV recovery by {name}");
+	let result = network
+		.get_node(name)?
+		.wait_log_line_count_with_timeout(
+			"Importing blocks retrieved using pov_recovery",
+			false,
+			options,
+		)
+		.await?;
+
+	assert!(result.success(), "Failed importing blocks using PoV recovery by {name}: {:?}", result);
+
+	log::info!("Test finished successfully");
+	Ok(())
+}
+
+async fn build_network_config() -> Result<NetworkConfig, anyhow::Error> {
+	let images = zombienet_sdk::environment::get_images_from_env();
+	log::info!("Using images: {images:?}");
+
+	// Network setup:
+	// - relaychain nodes:
+	// 	 - alice
+	// 	   - validator
+	// 	 - validator[0-7]
+	// 	   - validator
+	// 	   - synchronize only with alice
+	// - parachain nodes
+	//   - recovery-target
+	//     - full node
+	//   - collator-elastic
+	//     - collator which is the only one producing blocks
+	NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let r = r
+				.with_chain("rococo-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_resources(|resources| {
+					resources
+						.with_request_cpu(1)
+						.with_request_memory("2G")
+						.with_limit_cpu(2)
+						.with_limit_memory("4G")
+				})
+				.with_genesis_overrides(json!({
+					"configuration": {
+						"config": {
+							"scheduler_params": {
+								"num_cores": 4,
+								"max_validators_per_core": 1
+							},
+							"approval_voting_params": {
+								"max_approval_coalesce_count": 5
+							}
+						}
+					}
+				}))
+				// Have to set a `with_node` outside of the loop below, so that `r` has the right
+				// type.
+				.with_node(|node| node.with_name("alice").with_args(vec![]));
+
+			(0..8).fold(r, |acc, i| {
+				acc.with_node(|node| {
+					node.with_name(&format!("validator-{i}")).with_args(vec![
+						("-lruntime=debug,parachain=trace").into(),
+						("--reserved-only").into(),
+						("--reserved-nodes", "{{ZOMBIE:alice:multiaddr}}").into(),
+					])
+				})
+			})
+		})
+		.with_parachain(|p| {
+			p.with_id(PARA_ID)
+				.with_chain("elastic-scaling")
+				.with_registration_strategy(RegistrationStrategy::Manual)
+				.with_default_command("test-parachain")
+				.with_default_image(images.cumulus.as_str())
+				.with_default_resources(|resources| {
+					resources
+						.with_request_cpu(1)
```

Comment:
We could remove this right? Should have no effect on native runners? Or do you prefer to leave it for a potential comeback of k8s?

Reply:
Well, I had potential switch to `k8s` in back of my head when introducing all of these changes :) 
I hope it will be just provider switch in CI yaml and voila!
But TBH I haven't noticed this one. I think I wll add some comment that this is not relevant for native provider.

Reply:
Yes, at some point I think we will back to `k8s` but I think we can just have a comment here.
Thx!